Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UnitData.xml

```diff
@@ -2442,8 +2468,28 @@
     </CUnit>
     <CUnit id="AP_SpectreResourceEfficiency" parent="AP_Spectre">
         <AbilArray index="10" removed="1"/>
-        <BehaviorArray index="2" removed="1"/>
+        <BehaviorArray index="2" Link="AP_SpectrePermanentCloak"/>
+        <BehaviorArray index="3" removed="1"/>

```

Comment:
Try just cleaner `<BehaviorArray index="2" removed="1"/>`. If it doesn't work, rearrange the behaviors in `AP_Spectre` that the morph is the last